cinema/views.py

```diff
@@ -56,3 +95,42 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        movie = self.request.query_params.get("movie")
+        date = self.request.query_params.get("date")
+
+        if self.action in ("list", "retrieve"):
+            self.queryset = self.queryset.select_related("movie",
+                                                         "cinema_hall")
+
+        if movie:
+            self.queryset = self.queryset.filter(movie__id=movie)
+
+        if date:
+            self.queryset = self.queryset.filter(
+                show_time__date=datetime.datetime
+                .strptime(date, "%Y-%m-%d").date()
```

Comment:
this will work but keep in mind if we receive the wrongly-formatted data it'll be a ValueError